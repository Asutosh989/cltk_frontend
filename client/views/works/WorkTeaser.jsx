import injectTapEventPlugin from 'react-tap-event-plugin';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Divider from 'material-ui/Divider';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import CommunicationComment from 'material-ui/svg-icons/communication/comment';
import ActionInput from 'material-ui/svg-icons/action/input';
import ActionFavoriteBorder from 'material-ui/svg-icons/action/favorite-border';
// Needed for onTouchTap
// Check this repo:
// https://github.com/zilverline/react-tap-event-plugin
injectTapEventPlugin();
// Work Teaser
WorkTeaser = React.createClass({

  getChildContext() {
    return { muiTheme: getMuiTheme(baseTheme) };
  },

  propTypes: {
    work: React.PropTypes.object.isRequired
  },


  render() {
    let work = this.props.work;
    let work_url = "/works/" + work.slug ;

     return (
         <Card
					 className="work-teaser"
					 >

					 	<div className="card-meta-left">
							<IconButton
								tooltip="Add to Your Library"
					      tooltipPosition="top-center"
								className="icon-favorite-button"
								iconClassName="mdi mdi-book-open-variant"
								/>

							<div className="card-meta-items">
								<span className="card-meta card-meta-left-language">
										{Utils.capitalize(work.language)}
								</span>
								<span className="card-meta card-meta-left-date">
										{work.date}
								</span>
							</div>
					 	</div>

						<div className="work-teaser-authors">
							{work.authors.map((author, i) => {
								return <a
									key={i}
									href={"/authors/" + author.slug}
									className="work-teaser-author">
									<h4>
											{author.english_name}
											{author.original_name ?
												<span className="work-teaser-author-original-name">
													({author.original_name})
												</span>
											: ""}
									</h4>
								</a>
							})}
						</div>

            <a
							href={work_url}
							className="work-teaser-title"
							>
              <h3 >
								{work.english_title}
								{work.original_title ?
									<span className="work-teaser-original-title">
										{work.original_title}
									</span>
								: ""}
              </h3>
            </a>

						<div className="card-meta-bottom">
							<span className="card-meta meta-count-commentary">
								{work.countComments} Comments
							</span>
							<span className="card-meta meta-count-translations">
								{work.countTranslations} Translations
							</span>
							<span className="card-meta meta-count-entities">
								{work.countEntities} Entities
							</span>
							<span className="card-meta meta-count-annotations">
								{work.countAnnotations} Annotations
							</span>

						</div>



            {/*<Divider />
            <CardActions>
              <a href="#">
                <IconButton tooltip="Comment">
                  <CommunicationComment />
                </IconButton>
              </a>
              <a href="#">
                <IconButton tooltip="Favorite">
                  <ActionFavoriteBorder />
                </IconButton>
              </a>
              <a href="#">
                <IconButton tooltip="Other Formats">
                  <ActionInput />
                </IconButton>
              </a>
            </CardActions>*/}
          </Card>
      );
    }

});

WorkTeaser.childContextTypes = {
    muiTheme: React.PropTypes.object.isRequired,
};
