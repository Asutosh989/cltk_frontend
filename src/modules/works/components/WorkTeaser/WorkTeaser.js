import React from 'react';
import PropTypes from 'prop-types';
import autoBind from 'react-autobind';
import { Card } from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';



// Work Teaser
class WorkTeaser extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			isInShelf: false,
		};
		autoBind(this);
	}

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	}

	toggleShelf(isChecked) {
		this.setState({
			isInShelf: !this.state.isInShelf,
		});
	}

	render() {
		const work = this.props.work;
		const workUrl = `/works/${work.id}/${work.slug}`;

		let isInShelf = this.props.isInShelf;

		if (this.state.isInShelf) {
			isInShelf = this.state.isInShelf;
		}

		return (
			<Card
				className="work-teaser"
			>

				<div className="card-meta-left">
					<IconButton
						tooltip="Add to Your Library"
						tooltipPosition="top-center"
						className={`icon-favorite-button ${isInShelf ? 'in-user-shelf' : ''}`}
						iconClassName="mdi mdi-book-open-variant"
						onClick={this.toggleShelf.bind(this, isInShelf)}
					/>

					<div className="card-meta-items">
						<span className="card-meta card-meta-left-language">
							{work.language.title}
						</span>
						{work.date ?
							<span className="card-meta card-meta-left-date">
								{work.date}
							</span>
						:
							''
						}
						<span className="card-meta card-meta-left-corpus">
							{work.corpus.title}
						</span>
					</div>
				</div>

				<div className="work-teaser-authors">
					<div
						className="work-teaser-author"
					>

						<h4>
							{work.author.name}
							{work.author.original_name ?
								<span className="work-teaser-author-original-name">
									({work.author.original_name})
								</span>
								:
								''
							}
						</h4>
					</div>
				</div>

				<a
					href={workUrl}
					className="work-teaser-title"
				>
					<h3 >
						{work.english_title}
						{work.original_title ?
							<span className="work-teaser-original-title">
								{work.original_title}
							</span>
							:
							''
						}
					</h3>
				</a>

				<div className="card-meta-bottom">
					<span className="card-meta meta-count-commentary">
						{work.countComments || 0} Commentary
					</span>
					<span className="card-meta meta-count-translations">
						{work.countTranslations || 0} Translations
					</span>
					<span className="card-meta meta-count-annotations">
						{work.countAnnotations || 0} Annotations
					</span>
				</div>
			</Card>
		);
	}
}

WorkTeaser.childContextTypes = {
	muiTheme: PropTypes.object.isRequired,
};

WorkTeaser.propTypes = {
	work: PropTypes.object.isRequired,
};

export default WorkTeaser;
