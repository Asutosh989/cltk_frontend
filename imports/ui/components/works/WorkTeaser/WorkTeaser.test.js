
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import WorkTeaser from './WorkTeaser';

describe('WorkTeaser', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<WorkTeaser
						work={{
							_id: {
								_str: 'testId',
							},
							slug: 'test',
							isInShelf: false,
						}}
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
