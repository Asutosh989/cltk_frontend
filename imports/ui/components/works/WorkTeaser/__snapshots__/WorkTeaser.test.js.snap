// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkTeaser renders correctly 1`] = `
<div
  className="work-teaser"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      className="card-meta-left"
    >
      <button
        className="icon-favorite-button "
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="mdi mdi-book-open-variant"
            color="rgba(0, 0, 0, 0.87)"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          />
        </div>
      </button>
      <div
        className="card-meta-items"
      >
        
        
      </div>
    </div>
    <div
      className="work-teaser-authors"
    />
    <a
      className="work-teaser-title"
      href="/works/testId/test"
    >
      <h3>
        
      </h3>
    </a>
    <div
      className="card-meta-bottom"
    >
      <span
        className="card-meta meta-count-commentary"
      >
         Comments
      </span>
      <span
        className="card-meta meta-count-translations"
      >
         Translations
      </span>
      <span
        className="card-meta meta-count-annotations"
      >
         Annotations
      </span>
    </div>
  </div>
</div>
`;
